README.md

```diff
@@ -47,16 +56,149 @@ class LeonardoClient(leonardoBasePath: String) {
     leonardoApi.getAzureRuntime(workspaceId, runtimeName)
   }
 }
-
 ```
 
-## Building and running Leonardo
-Clone the repo.
-```
-$ git clone https://github.com/DataBiosphere/leonardo.git
-$ cd leonardo
+## Building and running `leonardo` locally
+
+### Clone the repo and submodules
+
+```sh
+git clone https://github.com/databiosphere/leonardo.git
+cd leonardo
 ```
 
+And as an aside, this repository uses git submodules. You will need to execute the following commands as well:
+
+```sh
+git submodule update --init --recursive
+```
+
+### Install `leonardo` dependencies
+
+The following tools are required to run `leonardo`:
+
+- vault
+- docker
+- google-cloud-sdk
+- cloud-sql-proxy
+- [gke-gcloud-auth-plugin](https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke)
+- java
+- sbt
+- go
+
+Please feel free to install each tool individually as you see fit for your environment, or
+you can follow along with this process to get your environment set up.
+Tool setup is facilitated through the use of [`brew`](https://brew.io). This allows us to have a little consistency
+across environments thanks to the `Brewfile.lock.json`
+
+- Install `brew`.
+  ```shell
+  /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+
+  # be sure to update your .[X]profile or .[X]shrc file with the following
+  # this assumes a default install location for `brew`
+  echo 'eval "$(/opt/homebrew/bin/brew shellenv)"' >> .zprofile
+  ```
+  After the `brew` install, validate that you have a working `brew` installation be executing `brew help`.
+  Feel free to [turn off Anonymous Analytics](https://docs.brew.sh/Analytics) with the following command:
+  ```sh
+  brew analytics off
+  ```
+
+- Following the `brew`-install, you should be able to install the necessary tools to setup the remaining dependencies.
```

Comment:
Ohhh this is really nice, I never even thought of checking if brew could build deterministic environments like poetry does 😅 this is super cool, thanks for making this for us ❤️